conans/model/conf.py

```diff
@@ -320,9 +320,10 @@ def get(self, conf_name, default=None, check_type=None, choices=None):
             if choices is not None and v not in choices:
                 raise ConanException(f"Unknown value '{v}' for '{conf_name}'")
             # Some smart conversions
-            if check_type is bool and not isinstance(v, bool):
-                # Perhaps, user has introduced a "false", "0" or even "off"
-                return str(v).lower() not in Conf.boolean_false_expressions
+            if check_type is bool and not (isinstance(v, bool) or str(v).lower() not in
+                                           Conf.boolean_false_expressions):
+                raise ConanException(f"[conf] {conf_name} must be a boolean-like object "
```

Comment:
I want to discuss this further with the team, I like the improvement in usage, but nontheless is still a breaking change


Also few comments:
* should this still handle the return explictly when it is a falsy value?
* we should add a test to cover this new case